spec/lib/gems/baw_workers/jobs/media/edge_cases_spec.rb

```diff
@@ -0,0 +1,141 @@
+# frozen_string_literal: true
+
+context 'with edge cases in media generation' do
+  include ExampleImageHelpers
+  include_context 'shared_test_helpers'
+
+  # fill our audio storage with a fixture
+
+  let(:uuid) { '7bb0c719-143f-4373-a724-8138219006d9' }
+  let(:recorded_date) { Time.zone.parse('2019-09-13T00:00:01+1000') }
+  let(:audio_payload) {
+    BawWorkers::Models::AudioRequest.new(
+      uuid: uuid,
+      start_offset: 7100,
+      end_offset: 7130,
+      channel: 0,
+      sample_rate: 22_050,
+      datetime_with_offset: recorded_date,
+      original_format: audio_file_bar_lt_metadata[:format],
+      original_sample_rate: audio_file_bar_lt_metadata[:sample_rate],
+      format: 'wav',
+      media_type: 'audio/wav'
+    )
+  }
+  let(:spectrogram_payload) {
+    BawWorkers::Models::SpectrogramRequest.new(
+      **audio_payload,
+      format: 'png',
+      media_type: 'image/png',
+      window: 512,
+      window_function: 'Hamming',
+      colour: 'g'
+    )
+  }
+  let!(:test_file) {
+    link_original_audio(
+      target: Fixtures.bar_lt_file,
+      uuid: uuid,
+      datetime_with_offset: recorded_date,
+      original_format: audio_file_bar_lt_metadata[:format]
+    )
+  }
+
+  before do
+    clear_audio_cache
+    clear_spectrogram_cache
+  end
+
+  after do
+    test_file.unlink
+  end
+
+  # https://github.com/QutEcoacoustics/baw-server/issues/527
+  context 'race conditions in interleaved audio & spectrograms jobs targeting the same segment' do
+    # we're running all jobs in the RSpec environment not on the workers for this test
+    pause_all_jobs
+
+    def perform_many_times(path, job, payload)
+      # 100 is a good size for general/CI testing - the test should take about 30 seconds
+      # I've manually tested sizes 200 & 300 too, after about 500 the tools start to timeout on my dev box
```

Comment:
A wise man once said that people don't run long test suites. If there is a way, adding some sort of flag to the test command to enable/disable slow tests may be a good option. Especially since testing baw-server can already take an extremely long time. It may be an idea to exclude these kind of long tests, and only have them run on the CI

Reply:
Good idea